src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -98,6 +110,19 @@ func parseParams(
 		return params, xerrors.NewInvalidParamsError(
 			fmt.Errorf(formatErrStr, QueryParam, err))
 	}
+
+	params.LookbackDuration = engineOpts.LookbackDuration()
+	if v := fetchOpts.LookbackDuration; v != nil {
+		params.LookbackDuration = *v
+	}
+
+	// Grafana lack of alert rule template variable expansion handling here (sigh)
+	// https://github.com/grafana/grafana/issues/6557
+	query = strings.ReplaceAll(query, "$__interval", shortDur(params.Step))
+	query = strings.ReplaceAll(query, "$__range", shortDur(params.LookbackDuration))
+	// Grafana 7.2 adds $__rate_interval which is still not expanded in alerting. According to documentation it is <= 4x step
+	query = strings.ReplaceAll(query, "$__rate_interval", shortDur(time.Duration(4)*params.Step))
```

Comment:
our alertmanager alerts should be fixed to drop these (I guess).  Or, maybe upstream could be convinced that having the macros is actually a good idea.